src/Adapter/SqlManager/QueryHandler/GetSqlRequestExecutionResultHandler.php

```diff
@@ -97,4 +99,34 @@ private function hideSensitiveData(array $records)
 
         return $records;
     }
+
+    /**
+     * Detect from list of sensitive attributes if function or alias are used in the sql query
+     * then add alias in the list of sensitives attributes to hide.
+     *
+     * @param string $query
+     *
+     * @return array
+     */
+    private function getSensitiveAttributes(string $query): array
+    {
+        $sensitiveAttributes = (new RequestSql())->attributes;
+        $parser = new PHPSQLParser();
+        $parsed = $parser->parse($query);
+        foreach ($parsed['SELECT'] as $selectField) {
+            if (is_array($selectField['alias'])) {
+                $alias = $selectField['alias']['name'];
+                while (is_array($selectField['sub_tree'])) {
```

Comment:
I'm always worried about `while` 😛 I'm afraid of infinite loop

Are you sure there is no way this would loop forever?

Reply:
yes sure :smile: